src/summary.rs

```diff
@@ -38,6 +38,7 @@ pub(crate) struct SummaryUpdateFields {
     pub(crate) new_vote_count: u64,
     pub(crate) new_reward: Rewards,
     pub(crate) new_parsed_blocks: BlockNumber,
+    pub(crate) maybe_updated_user_space_pledged: Option<ByteSize>,
```

Comment:
When would it be `None`? Or why is `Default` needed on `SummaryUpdateFields`?

Reply:
So, it would be `None` when we are updating other fields like `new_vote_count`, `new_reward` etc after reading from subscribed data stream.

if it is not `Option` type, we would have to query the current space pledged every time we want to update any data inside summary file due to the way the summary file is handled at present.